### PR TITLE
dev query find committee by privilege [ch6198]

### PR DESCRIPTION
**Pull request by Raulnet**
Fix bug display dropdown for single committee and CSS push right arrown of dropdown [ch6198]

### CODE REVIEW

File: src/Controller/EnMarche/AdherentController.php [hunk 1]
```diff
@@ -294,12 +296,10 @@ public function listMyCommitteesAction(string $noResultMessage = null): Response
...
     }
 
-    public function listCommitteesAlAction(): Response
+    public function listCommitteesAlAction(CommitteeMembershipRepository $repository): Response
     {
-        $manager = $this->get('app.committee.manager');
-
         return $this->render('adherent/list_my_committees_al.html.twig', [
-            'committees' => $manager->getAdherentCommittees($this->getUser()),
+            'committees' => $repository->findCommitteeByPrivilege($this->getUser(), CommitteeMembership::getHostPrivileges()),
```
Comment (thomasbisignani) [inline src/Controller/EnMarche/AdherentController.php]: Can you multiple line here please ?

File: src/Repository/CommitteeMembershipRepository.php [hunk 2]
```diff
@@ -417,4 +417,21 @@ public function findCommitteesUuid(array $criteria): array
             return $uuid->toString();
         }, array_column($qb->getQuery()->getArrayResult(), 'uuid'));
     }
+
+    public function findCommitteeByPrivilege(Adherent $adherent, array $privilege): array
+    {
+        // Prevent SQL query if the adherent doesn't follow any committees yet.
+        if (0 === \count($memberships = $adherent->getMemberships())) {
+            return [];
+        }
```
Comment (thomasbisignani) [inline src/Repository/CommitteeMembershipRepository.php]: Not a big deal but can you add a new line after this `if` please ?

File: src/Repository/CommitteeMembershipRepository.php [hunk 3]
```diff
@@ -417,4 +417,21 @@ public function findCommitteesUuid(array $criteria): array
...
+        $qb = $this->createQueryBuilder('cm');
+        $qb->select('c.id, c.name, c.slug')
```
Comment (thomasbisignani) [inline src/Repository/CommitteeMembershipRepository.php]: Can you use ` ->addSelect` instead ? 

And indent like : 

```php
$qb = $this
    ->createQueryBuilder('cm')
    ->addSelect('...')
    ->innerJoin('...')
```

File: src/Repository/CommitteeMembershipRepository.php [hunk 4]
```diff
@@ -417,4 +417,21 @@ public function findCommitteesUuid(array $criteria): array
             return $uuid->toString();
         }, array_column($qb->getQuery()->getArrayResult(), 'uuid'));
     }
+
+    public function findCommitteeByPrivilege(Adherent $adherent, array $privilege): array
```
Comment (thomasbisignani) [inline src/Repository/CommitteeMembershipRepository.php]: Can you rename this method to `findCommitteesByPrivilege` ?

Comment (Raulnet) [reply src/Repository/CommitteeMembershipRepository.php]: @thomasbisignani done

File: src/Repository/CommitteeMembershipRepository.php [hunk 5]
```diff
@@ -417,4 +417,24 @@ public function findCommitteesUuid(array $criteria): array
...
+    public function findCommitteesByPrivilege(Adherent $adherent, array $privilege): array
+    {
+        // Prevent SQL query if the adherent doesn't follow any committees yet.
+        if (0 === \count($adherent->getMemberships())) {
+            return [];
+        }
+
+        $qb = $this->createQueryBuilder('cm');
+
+        return $qb
```
Comment (thomasbisignani) [inline src/Repository/CommitteeMembershipRepository.php]: You can remove the `$qb` variable and return directly : 

```php 
return $this
    ->createQueryBuilder('cm')
    ->addSelect('c.id, c.name, c.slug')
    // ...
```

Comment (Raulnet) [reply src/Repository/CommitteeMembershipRepository.php]: i need to use ```$qb``` for the ```->where($qb->expr()->in('cm.privilege', $privilege))```

Comment (thomasbisignani) [reply src/Repository/CommitteeMembershipRepository.php]: Yes okay :wink:

**Pull request merged**